Loading page hashes has deadlocks if comics are also being processed simultaneously.

The error being reported is:
```
  org.springframework.dao.CannotAcquireLockException: could not execute statement [ERROR: deadlock detected
  Detail: Process 5068 waits for ShareLock on transaction 625343; blocked by process 18672.
Process 18672 waits for ShareLock on transaction 625371; blocked by process 5068.
  Hint: See server log for query details.
  Where: while updating tuple (11,22) in relation "comic_pages"] [update comic_pages set page_number=? where id=?]; SQL [update comic_pages set page_number=? where id=?]
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:283)
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:244)
	at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:565)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.processCommit(AbstractPlatformTransactionManager.java:794)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:757)
	at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:152)
	at org.springframework.batch.core.step.tasklet.TaskletStep$2.doInChunkContext(TaskletStep.java:255)
	at org.springframework.batch.core.scope.context.StepContextRepeatCallback.doInIteration(StepContextRepeatCallback.java:82)
	at org.springframework.batch.repeat.support.RepeatTemplate.getNextResult(RepeatTemplate.java:369)
	at org.springframework.batch.repeat.support.RepeatTemplate.executeInternal(RepeatTemplate.java:206)
	at org.springframework.batch.repeat.support.RepeatTemplate.iterate(RepeatTemplate.java:140)
	at org.springframework.batch.core.step.tasklet.TaskletStep.doExecute(TaskletStep.java:240)
	at org.springframework.batch.core.step.AbstractStep.execute(AbstractStep.java:229)
	at org.springframework.batch.core.job.SimpleStepHandler.handleStep(SimpleStepHandler.java:153)
	at org.springframework.batch.core.job.AbstractJob.handleStep(AbstractJob.java:418)
	at org.springframework.batch.core.job.SimpleJob.doExecute(SimpleJob.java:132)
	at org.springframework.batch.core.job.AbstractJob.execute(AbstractJob.java:317)
	at org.springframework.batch.core.launch.support.SimpleJobLauncher$1.run(SimpleJobLauncher.java:157)
	at java.base/java.lang.Thread.run(Thread.java:842)
Caused by: org.hibernate.exception.LockAcquisitionException: could not execute statement [ERROR: deadlock detected
  Detail: Process 5068 waits for ShareLock on transaction 625343; blocked by process 18672.
Process 18672 waits for ShareLock on transaction 625371; blocked by process 5068.
  Hint: See server log for query details.
  Where: while updating tuple (11,22) in relation "comic_pages"] [update comic_pages set page_number=? where id=?]
	at org.hibernate.dialect.PostgreSQLDialect.lambda$buildSQLExceptionConversionDelegate$1(PostgreSQLDialect.java:1015)
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:58)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:108)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197)
	at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.performNonBatchedMutation(AbstractMutationExecutor.java:108)
	at org.hibernate.engine.jdbc.mutation.internal.MutationExecutorSingleNonBatched.performNonBatchedOperations(MutationExecutorSingleNonBatched.java:40)
	at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.execute(AbstractMutationExecutor.java:52)
	at org.hibernate.persister.collection.OneToManyPersister.writeIndex(OneToManyPersister.java:260)
	at org.hibernate.persister.collection.OneToManyPersister.insertRows(OneToManyPersister.java:182)
	at org.hibernate.action.internal.CollectionUpdateAction.execute(CollectionUpdateAction.java:86)
	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:632)
	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:499)
	at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:363)
	at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:41)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)
	at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1403)
	at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:484)
	at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:2319)
	at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:1976)
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:439)
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:169)
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:267)
	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:101)
	at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:561)
	... 16 more
Caused by: org.postgresql.util.PSQLException: ERROR: deadlock detected
  Detail: Process 5068 waits for ShareLock on transaction 625343; blocked by process 18672.
Process 18672 waits for ShareLock on transaction 625371; blocked by process 5068.
  Hint: See server log for query details.
  Where: while updating tuple (11,22) in relation "comic_pages"
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2713)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2401)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:368)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:498)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:415)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:190)
	at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:152)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:194)
	... 36 more
```
or:
``` 
  org.springframework.transaction.UnexpectedRollbackException: Transaction silently rolled back because it has been marked as rollback-only
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.processCommit(AbstractPlatformTransactionManager.java:803)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:757)
	at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:152)
	at org.springframework.batch.core.step.tasklet.TaskletStep$2.doInChunkContext(TaskletStep.java:255)
	at org.springframework.batch.core.scope.context.StepContextRepeatCallback.doInIteration(StepContextRepeatCallback.java:82)
	at org.springframework.batch.repeat.support.RepeatTemplate.getNextResult(RepeatTemplate.java:369)
	at org.springframework.batch.repeat.support.RepeatTemplate.executeInternal(RepeatTemplate.java:206)
	at org.springframework.batch.repeat.support.RepeatTemplate.iterate(RepeatTemplate.java:140)
	at org.springframework.batch.core.step.tasklet.TaskletStep.doExecute(TaskletStep.java:240)
	at org.springframework.batch.core.step.AbstractStep.execute(AbstractStep.java:229)
	at org.springframework.batch.core.job.SimpleStepHandler.handleStep(SimpleStepHandler.java:153)
	at org.springframework.batch.core.job.AbstractJob.handleStep(AbstractJob.java:418)
	at org.springframework.batch.core.job.SimpleJob.doExecute(SimpleJob.java:132)
	at org.springframework.batch.core.job.AbstractJob.execute(AbstractJob.java:317)
	at org.springframework.batch.core.launch.support.SimpleJobLauncher$1.run(SimpleJobLauncher.java:157)
	at java.base/java.lang.Thread.run(Thread.java:842)
```

--- a/comixed-adaptors/src/main/java/org/comixedproject/adaptors/comicbooks/ComicBookAdaptor.java
+++ b/comixed-adaptors/src/main/java/org/comixedproject/adaptors/comicbooks/ComicBookAdaptor.java
@@ -21,8 +21,6 @@ package org.comixedproject.adaptors.comicbooks;
 import java.io.*;
 import java.util.List;
 import java.util.Optional;
-import java.util.Set;
-import java.util.concurrent.ConcurrentSkipListSet;
 import lombok.NonNull;
 import lombok.extern.log4j.Log4j2;
 import org.apache.commons.compress.utils.FileNameUtils;
@@ -63,8 +61,6 @@ public class ComicBookAdaptor {
   @Autowired private ComicMetadataContentAdaptor comicMetadataContentAdaptor;
   @Autowired private FileAdaptor fileAdaptor;
 
-  private Set<String> openFiles = new ConcurrentSkipListSet<>();
-
   /**
    * Creates a new comic. Determines the archive type for the underlying file.
    *
@@ -90,50 +86,41 @@ public class ComicBookAdaptor {
    * @throws AdaptorException if an error occurs while loading the comic book file
    */
   public void load(final ComicBook comicBook, final ContentAdaptorRules rules)
-      throws AdaptorException, InterruptedException {
-    @NonNull final String filename = comicBook.getComicDetail().getFilename();
-    synchronized (this.openFiles) {
-      while (this.openFiles.contains(filename)) {
-        this.openFiles.wait();
-      }
-      try {
-        this.openFiles.add(filename);
-        log.trace(
-            "Getting archive adaptor for comic book file: id={} rule={}",
-            comicBook.getId(),
-            rules.toString());
-        final ArchiveAdaptor archiveAdaptor = this.fileTypeAdaptor.getArchiveAdaptorFor(filename);
-        log.trace("Opening comic book file");
-        final ArchiveReadHandle readHandle = archiveAdaptor.openArchiveForRead(filename);
-        log.trace("Loading comic book file entries");
-        final List<ComicArchiveEntry> entries = archiveAdaptor.getEntries(readHandle);
-        for (int index = 0; index < entries.size(); index++) {
-          final ComicArchiveEntry entry = entries.get(index);
-          if (entry.getSize() > 0) {
-            log.trace("Loading entry content: {}", entry.getFilename());
-            final byte[] content = archiveAdaptor.readEntry(readHandle, entry.getFilename());
-            if (content.length > 0) {
-              log.trace("Getting content adaptor for entry: {}", entry.getFilename());
-              final ContentAdaptor adaptor = this.fileTypeAdaptor.getContentAdaptorFor(content);
-              if (adaptor != null) {
-                log.trace("Invoking content adaptor");
-                adaptor.loadContent(comicBook, entry.getFilename(), content, rules);
-              }
-            } else {
-              log.trace("Content contains no data");
+      throws AdaptorException {
+    try {
+      @NonNull final String filename = comicBook.getComicDetail().getFilename();
+      log.trace(
+          "Getting archive adaptor for comic book file: id={} rule={}",
+          comicBook.getId(),
+          rules.toString());
+      final ArchiveAdaptor archiveAdaptor = this.fileTypeAdaptor.getArchiveAdaptorFor(filename);
+      log.trace("Opening comic book file");
+      final ArchiveReadHandle readHandle = archiveAdaptor.openArchiveForRead(filename);
+      log.trace("Loading comic book file entries");
+      final List<ComicArchiveEntry> entries = archiveAdaptor.getEntries(readHandle);
+      for (int index = 0; index < entries.size(); index++) {
+        final ComicArchiveEntry entry = entries.get(index);
+        if (entry.getSize() > 0) {
+          log.trace("Loading entry content: {}", entry.getFilename());
+          final byte[] content = archiveAdaptor.readEntry(readHandle, entry.getFilename());
+          if (content.length > 0) {
+            log.trace("Getting content adaptor for entry: {}", entry.getFilename());
+            final ContentAdaptor adaptor = this.fileTypeAdaptor.getContentAdaptorFor(content);
+            if (adaptor != null) {
+              log.trace("Invoking content adaptor");
+              adaptor.loadContent(comicBook, entry.getFilename(), content, rules);
             }
           } else {
-            log.trace("Entry contains no data");
+            log.trace("Content contains no data");
           }
+        } else {
+          log.trace("Entry contains no data");
         }
-        log.trace("Closing comic book file");
-        archiveAdaptor.closeArchiveForRead(readHandle);
-      } catch (AdaptorException | ArchiveAdaptorException | ContentAdaptorException error) {
-        throw new AdaptorException("Failed to load comic book file", error);
-      } finally {
-        this.openFiles.remove(filename);
-        this.openFiles.notifyAll();
       }
+      log.trace("Closing comic book file");
+      archiveAdaptor.closeArchiveForRead(readHandle);
+    } catch (AdaptorException | ArchiveAdaptorException | ContentAdaptorException error) {
+      throw new AdaptorException("Failed to load comic book file", error);
     }
   }
 

